GooglePlayInstant/Editor/AppBundleBuilder.cs

```diff
@@ -0,0 +1,202 @@
+        }
+
+        private static void DisplayProgress(string info, float progress)
+        {
+            Debug.LogFormat("{0}...", info);
+            if (!PlayInstantBuilder.IsHeadlessMode())
+            {
+                EditorUtility.DisplayProgressBar("Building App Bundle", info, progress);
+            }
+        }
+
+        private static void LogError(string errorType, string errorMessage)
+        {
+            if (!PlayInstantBuilder.IsHeadlessMode())
+            {
+                EditorUtility.ClearProgressBar();
+            }
+
+            PlayInstantBuilder.LogError(string.Format("{0} failed: {1}", errorType, errorMessage));
+        }
+
+        private static void ConvertFiles(DirectoryInfo source, DirectoryInfo destination)
+        {
+            // Copy files in the root directory.
+            foreach (var fileInfo in source.GetFiles())
+            {
+                var fileName = fileInfo.Name;
+                if (fileName == "AndroidManifest.xml")
+                {
+                    var subdirectory = destination.CreateSubdirectory("manifest");
+                    fileInfo.CopyTo(Path.Combine(subdirectory.FullName, fileName));
+                }
+                else if (fileName == "resources.pb")
+                {
+                    fileInfo.CopyTo(Path.Combine(destination.FullName, fileName));
+                }
+                else if (fileName.EndsWith("dex"))
+                {
+                    var subdirectory = destination.CreateSubdirectory("dex");
+                    fileInfo.CopyTo(Path.Combine(subdirectory.FullName, fileName));
+                }
+                else
+                {
+                    var subdirectory = destination.CreateSubdirectory("root");
+                    fileInfo.CopyTo(Path.Combine(subdirectory.FullName, fileName));
+                }
+            }
+
+            // Copy all other files.
+            foreach (var directoryInfo in source.GetDirectories())
+            {
+                var directoryName = directoryInfo.Name;
+                switch (directoryName)
+                {
+                    case "META-INF":
+                        // Skip files like MANIFEST.MF
+                        break;
+                    case "assets":
+                    case "lib":
+                    case "res":
+                        CopyDirectory(directoryInfo, destination.CreateSubdirectory(directoryName));
+                        break;
+                    default:
+                        var subdirectory = destination.CreateSubdirectory("root");
+                        CopyDirectory(directoryInfo, subdirectory.CreateSubdirectory(directoryName));
+                        break;
+                }
+            }
+        }
+
+        private static void CopyDirectory(DirectoryInfo source, DirectoryInfo destination)
+        {
+            if (!destination.Exists)
+            {
+                destination.Create();
```

Comment:
I don't think you need to check if the directory exists before creating it. DirectoryInfo.Create should already do nothing if the directory exists.

Reply:
Good call. I don't check this in similar situations below.